src/item_box.py

```diff
@@ -33,6 +34,36 @@ def __init__(self, parent, item):
         self.color_button.get_style_context().add_provider(
             self.provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION)
         self.update_color()
+        self.drag_source = Gtk.DragSource.new()
+        self.drag_source.set_actions(Gdk.DragAction.COPY)
+
+
+        self.drop_source = Gtk.DropTarget.new(str, Gdk.DragAction.COPY)
+        self.drop_source.key = item.key
+        self.drop_source.connect('drop', self.on_dnd_drop)
+        self.drag_source.connect('prepare', self.on_dnd_prepare)
+
+        self.add_controller(self.drag_source)
+        self.add_controller(self.drop_source)
+
+    def on_dnd_drop(self, drop_target, value, x, y):
+        # Handle the dropped data here
+        self.parent.datadict = utilities.swap_key_positions(
+        self.parent.datadict, drop_target.key, value)
+        self.parent.item_boxes = utilities.swap_key_positions(
+        self.parent.item_boxes, drop_target.key, value)
+        self.parent.item_menu = utilities.swap_key_positions(
+        self.parent.item_menu, drop_target.key, value)
+        for key, item in self.parent.item_menu.items():
+            self.parent.main_window.list_box.remove(item)
+            self.parent.main_window.list_box.append(item)
+        graphs.reload(self.parent)
+
+
+    def on_dnd_prepare(self, drag_source, x, y):
+        data = self.item.key
```

Comment:
would `utilities.shorten_label(self.item.name)` suffice to change to the tooltip to the actual name or is the key required for this to function?